### PR TITLE
Customize kafka admin client

### PR DESCRIPTION
**Pull request by IgnisDeusZero**
Customize kafka admin client as in https://github.com/vektonn/vektonn/pull/13

### CODE REVIEW

File: src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs [hunk 1]
```diff
@@ -20,18 +20,29 @@ public KafkaTopicCreator(ILog log, KafkaProducerConfig kafkaProducerConfig)
...
-            {
-                BootstrapServers = string.Join(",", kafkaProducerConfig.BootstrapServers),
-            };
+            var adminClientConfig = kafkaTopicCreationConfig.GetAdminClientConfig(kafkaProducerConfig.BootstrapServers);
+
+            adminClient = BuildAdmin(adminClientConfig);
+        }
 
-            var adminClientBuilder = new AdminClientBuilder(adminClientConfig)
+        private IAdminClient BuildAdmin(AdminClientConfig adminClientConfig)
```
Comment (kvadros) [inline src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs:28]: `BuildAdmin` -> `BuildAdminClient`

File: src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs [hunk 2]
```diff
@@ -20,18 +20,29 @@ public KafkaTopicCreator(ILog log, KafkaProducerConfig kafkaProducerConfig)
...
+        {
+            return new AdminClientBuilder(adminClientConfig)
                 .SetErrorHandler(
-                    (_, error) => log.Error($"ConfluentAdminClient error: {error.ToPrettyJson()}"))
+                    (_, error) =>
+                    {
+                        if (error.IsFatal)
+                        {
+                            LogExtensions.Error(log, $"ConfluentAdminClient error: {error.ToPrettyJson()}");
```
Comment (kvadros) [inline src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs:36]: Can we convert static method invocations from `LogExtensions` to extension method calls? (And do it everywhere in solution)

File: src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs (hunk 1 above)
Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs]: https://github.com/vektonn/vektonn/pull/15/commits/e7fa1fe93ee5570b4f98f8a8e6c2739516ad6046

File: src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs (hunk 2 above)
Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaTopicCreator.cs]: https://github.com/vektonn/vektonn/pull/15/commits/22a404eb9668557f02d6cb9126aa38f07f0eb1ad

Done

File: src/Vektonn.DataSource/Kafka/KafkaConsumer.cs [hunk 3]
```diff
@@ -81,20 +81,20 @@ public async Task RunAsync(CancellationToken cancellationToken)
                     {
                         if (error.IsFatal)
                         {
-                            LogExtensions.Error(this.log, $"ConfluentConsumer error: {error.ToPrettyJson()}");
+                            this.log.Error($"ConfluentConsumer error: {error.ToPrettyJson()}");
```
Comment (kvadros) [inline src/Vektonn.DataSource/Kafka/KafkaConsumer.cs:84]: Don't use `this` qualifier here, just `log`.

Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaConsumer.cs]: https://github.com/vektonn/vektonn/pull/15/commits/9d27d19a072fa605366ddb4599bd207e8a319409

**Pull request merged**